src/modules/mc_att_control/mc_att_control_main.cpp

```diff
@@ -842,29 +839,35 @@ MulticopterAttitudeControl::control_attitude(float dt)
 	q.normalize();
 	qd.normalize();
 
-	/* calculate reduced attitude which we would command if we about the vehicle's yaw */
+	/* calculate reduced desired attitude neglecting vehicle's yaw to prioritize roll and pitch */
 	Vector3f e_z = q.dcm_z();
 	Vector3f e_z_d = qd.dcm_z();
 	Quatf qd_red(e_z, e_z_d);
-	qd_red *= q;
+
+	if (Quatf(qd_red - Quatf(0, 1, 0, 0)).norm() > 1e-5f) {
```

Comment:
Use norm squared to avoid calling `sqrt`?

And maby I'd even add something like you added as commit message as code comment:
```
Corner case with arbitrary yaw but and 180 degree roll or pitch:
Reduced attitude control calculation always rotates around roll
because there's no right choice when neglecting yaw. In that small
corner case it's better to just use full attitude contol and hence
rotate around the most efficient roll/pitch combination.
```

Reply:
👍 Thanks for reviewing and the hints. You catched the one line which I'm still not pleased with. I think it's still better to have no condition at all than this condition. Everything works and is stable without this condition but if you make up a special case where the vehicle needs to rotate exactly (machine accuracy close) 180 degree around roll or pitch but no rotation around yaw the result is not the optimal control law. I will loose some sleep over it and hopefully find a good solution.

Reply:
Found a better solution! What do you think?